src/components/shared/chart/Canvas.js

```diff
@@ -41,6 +40,19 @@ type State<HoveredItem> = {
   pageY: CssPixels,
 };
 
+export type ChartCanvasScale = {
+  // Always equal to devicePixelRatio
+  cssToDeviceScale: number,
+  // 1 if scaleCtxToCssPixels is true, otherwise equal to cssToDeviceScale
+  cssToUserScale: number,
+};
```

Comment:
Is it useful to have both of them? I understand the rationale but I see it's used just to compare both to detect errors. Maybe just `cssToUserScale` is useful, because that's the one that actually changes when `scaleCtxToCssPixels` changes? But also `scaleCtxToCssPixels` is set by the caller anyway so the caller knows about that.
All in all I'm not convinced by adding this structure compared to just keep using `devicePixelRatio` directly. But I'm not against either so we can keep it :-)

Reply:
I'm undecided. Their "usefulness" comes into play when somebody reads the `drawFilter` method without having read the rest of the file; you can see at the start of the method which space you're rendering into.